odilia/src/state.rs

```diff
@@ -36,6 +40,30 @@ pub struct ScreenReaderState {
 	pub cache: Arc<Cache>,
 }
 
+pub trait CacheProvider {
+	fn cache(&self) -> Arc<Cache>;
+}
+impl CacheProvider for Arc<ScreenReaderState> {
+	fn cache(&self) -> Arc<Cache> {
+		Arc::clone(&self.cache)
+	}
+}
+
+impl AsyncTryFrom<Arc<ScreenReaderState>> for Speech {
```

Comment:
Not sure I understand.

Reply:
why is a special async TryFrom trait required here? why are we implementing it in this way? why would we want to convert between the bigger state type and one of its fields, treating one like the other or as if it can be converted from the other?

Reply:
Yes, in this specific case it works fine.

This is also used for polling functions like generating the `CacheEvent<E>` for a given event. Since specialization does not exist, I need to choose one trait that can be applied to all parameters. There is little overhead to this since I directly return a `Ready<_>` future.